### PR TITLE
[SNC-392] compile option for tests

### PR DESCRIPTION
**Pull request by davidmdm**
This PR adds two properties to test cases within the agent test-runner.

```
compile: bool
pipeline_parameters: map[string]any
```

These two properties trigger the compilation of the given test input, and merges a new key: `_compile_` onto the test input.
This allows for users to write tests that will be compiled like in production.

The agent by design does not implement compiling or depend on any given compiler but instead accepts a single compile function with the following signature:

```go
func(input []byte, parameters map[string]any) (output []byte, err error)
```

---

Like other test properties `input`, `meta`, and `decision`, `compile` and `pipeline_parameters` are inherited and extended by subcases.

### CODE REVIEW

File: cpa/tester/compiler_policies/parameters/parameter.rego [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org
+
+policy_name["parameters"]
+
+enable_rule["assert_params"]
+
+# for this test, the mock test compiler will inject the pipeline parameters as the _compiled_ value
```
Comment (sagar-connect) [inline cpa/tester/compiler_policies/parameters/parameter.rego:7]: This sounds confusing to me, as we are comparing two different things (parameters and compiled config).

Should we instead have an "expected_compiled" to compare expected and actual compiled value instead.

**Review by sagar-connect — APPROVED** [review]

File: cpa/tester/compiler_policies/parameters/parameter.rego (hunk 1 above)
Comment (sagar-connect) [reply cpa/tester/compiler_policies/parameters/parameter.rego:7]: I understand it is probably the only and a smart way to test both compiler option and parameter parsing.

Comment (davidmdm) [reply cpa/tester/compiler_policies/parameters/parameter.rego:7]: So, the agent doesn't have a compiler built-in to it. It accepts the compilation function, and so each test provides the compilation logic based on the input and pipeline parameters. 

This test just wants to prove that pipeline parameters are indeed passed and merged as expected.

To test this, the "compiler" just forwards the pipeline parameters to `_compiled_` and we can make assertions using a rule that the expected_parameters match the _compiled_ parameters.

Does that make sense?

File: cpa/policy.go [hunk 2]
```diff
@@ -30,10 +30,7 @@ func (policy Policy) Modules() map[string]*ast.Module {
 // Eval will run native OPA query against your document, input, and apply any evaluation options.
 // It returns raw OPA expression values.
 func (policy Policy) Eval(ctx context.Context, query string, input interface{}, opts ...EvalOption) (interface{}, error) {
-	input, err := internal.ConvertYAMLMapKeyTypes(input)
-	if err != nil {
-		return nil, fmt.Errorf("invalid value: %w", err)
-	}
+	input = internal.ConvertYAMLMapKeyTypes(input)
```
[the remark this answers is not in the document]
Comment (davidmdm) [reply cpa/policy.go:33]: The function signature changed. It does not return an error anymore

**Review by rooneyshuman — APPROVED** [review]

**Pull request merged**